src/blockdata/witness.rs

```diff
@@ -282,18 +281,69 @@ impl serde::Serialize for Witness {
     where
         S: serde::Serializer,
     {
-        let vec: Vec<_> = self.to_vec();
-        serde::Serialize::serialize(&vec, serializer)
+        use hashes::hex::ToHex;
+        use serde::ser::SerializeSeq;
```

Comment:
I'm curious, why are these `use` statements put here rather than the top of the file?

Reply:
woops, I had a stale view of this page when I wrote my answer, I did not see @dr-orlovsky's answer at that time.